torch/nn/modules/transformer.py

```diff
@@ -368,7 +374,7 @@ def __init__(self, decoder_layer, num_layers, norm=None):
 
     def forward(self, tgt: Tensor, memory: Tensor, tgt_mask: Optional[Tensor] = None,
                 memory_mask: Optional[Tensor] = None, tgt_key_padding_mask: Optional[Tensor] = None,
-                memory_key_padding_mask: Optional[Tensor] = None, tgt_is_causal: bool = False,
+                memory_key_padding_mask: Optional[Tensor] = None, tgt_is_causal: Optional[bool] = None,
```

Comment:
For posterity, from offline discussion 

`bool` --> `Optional[bool]` was for consistency with `is_causal` in `TransformerEncoder` and is not BC-breaking since this kwarg was not in the 2.0 release


Separately, I have a question here -- why do we not do the same for `memory_is_causal`?

Reply:
@janEbert submitted the original diff that added tgt_is_causal as a Boolean, this ratifies this change in TransformerDecoder further and aligns it with TransformerEncoder.

@janEbert do you have thoughts on the utility of adding the same causal Optional[bool] kwarg `memory_is_causal` ?

Reply:
My thinking was that a user pretty much never supplies a causal memory mask and automatically checking for it would be a waste 99 % of the time. :)

Reply:
We might ask whether we should introduce the kwarg memory_is_causal : Optional[bool], but then skip detect and just  `if memory_is_causal is None: memory_is_causal = False` with a comment to the effect that we're not checking because memory mask is not commonly a causal mask.

Or is that just forcing consistency in form on something that in practice doesn't have consistency and we should just add a sentence to the doc string why memory_mask has no corresponding is_causal?